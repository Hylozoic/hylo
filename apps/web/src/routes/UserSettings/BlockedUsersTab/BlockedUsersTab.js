import React, { useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import { useViewHeader } from 'contexts/ViewHeaderContext'
import classes from './BlockedUsersTab.module.scss'
import Loading from 'components/Loading'

function BlockedUsersTab ({ blockedUsers, unBlockUser, loading }) {
  const { t } = useTranslation()

  const { setHeaderDetails } = useViewHeader()
  useEffect(() => {
    setHeaderDetails({
      title: t('Blocked Users'),
      icon: '',
      info: '',
      search: false
    })
  }, [])

  if (loading) return <Loading />

  return (
    <div>
      {blockedUsers && blockedUsers.map(blockedUser =>
        <UnBlockUserControl
          blockedUser={blockedUser}
          unBlockUser={unBlockUser}
          key={blockedUser.id}
        />)}
      {(!blockedUsers || blockedUsers.length === 0) && <div className={classes.noBlockedUsers}>{t('No blocked users')}</div>}
    </div>
  )
}

export function UnBlockUserControl ({ blockedUser, unBlockUser }) {
  const unBlockUserFun = () => unBlockUser(blockedUser.id)
  const { t } = useTranslation()

  return (
    <div className={classes.unblockUserControl}>
      <div className={classes.row}>
        <div className={classes.name}>{blockedUser.name}</div>
        <div onClick={unBlockUserFun} className={classes.unblockButton}>{t('Unblock')}</div>
      </div>
    </div>
  )
}

export default BlockedUsersTab
