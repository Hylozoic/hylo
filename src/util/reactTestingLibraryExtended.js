import React from 'react'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from 'store/reducers'
import createMiddleware from 'store/middleware'

export function generateStore (history, initialState) {
  return createStore(rootReducer, initialState, createMiddleware(history))
}

export const AllTheProviders = (store) => ({ children }) => {
  return (
    <Provider store={store || createStore(() => {})}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
}

// re-export everything

const rtl = require('react-testing-library')
const customRender = (ui, options, providersFunc) =>
  rtl.render(ui, { wrapper: providersFunc, ...options })

module.exports = {
  ...rtl,
  render: customRender
}
