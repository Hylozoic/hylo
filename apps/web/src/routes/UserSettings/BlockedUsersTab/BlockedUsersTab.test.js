import React from 'react'
import { render, screen, fireEvent, AllTheProviders } from 'util/testing/reactTestingLibraryExtended'
import BlockedUsersTab, { UnBlockUserControl } from './BlockedUsersTab'

describe('BlockedUsersTab', () => {
  it('renders a list of UnBlockUserControls', () => {
    const blockedUsers = [
      { id: 1, name: 'User 1' },
      { id: 2, name: 'User 2' },
      { id: 3, name: 'User 3' },
      { id: 4, name: 'User 4' }
    ]
    render(<BlockedUsersTab blockedUsers={blockedUsers} />)

    blockedUsers.forEach(user => {
      expect(screen.getByText(user.name)).toBeInTheDocument()
    })

    expect(screen.getAllByText('Unblock')).toHaveLength(4)
  })

  it('shows loading state when loading', () => {
    render(<BlockedUsersTab loading />)
    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument()
  })
})

describe('UnBlockUserControl', () => {
  it('renders correctly and handles unblock action', () => {
    const blockedUser = {
      id: 'test-user-id',
      name: 'Test User'
    }
    const unBlockUser = jest.fn()

    render(
      <UnBlockUserControl blockedUser={blockedUser} unBlockUser={unBlockUser} />
    )

    expect(screen.getByText('Test User')).toBeInTheDocument()
    const unblockButton = screen.getByText('Unblock')
    expect(unblockButton).toBeInTheDocument()

    fireEvent.click(unblockButton)
    expect(unBlockUser).toHaveBeenCalledWith('test-user-id')
  })
})
