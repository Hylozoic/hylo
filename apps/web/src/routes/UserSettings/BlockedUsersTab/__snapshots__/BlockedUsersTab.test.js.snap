// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockedUsersTab renders a list of UnBlockUserControls 1`] = `
<div>
  <UnBlockUserControl
    blockedUser={
      Object {
        "id": 1,
      }
    }
    key="1"
  />
  <UnBlockUserControl
    blockedUser={
      Object {
        "id": 2,
      }
    }
    key="2"
  />
  <UnBlockUserControl
    blockedUser={
      Object {
        "id": 3,
      }
    }
    key="3"
  />
  <UnBlockUserControl
    blockedUser={
      Object {
        "id": 4,
      }
    }
    key="4"
  />
</div>
`;
